src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -537,8 +537,13 @@ private void stopConnection(String errorMessage) {
      * @param args   The arguments to be passed to the method.
      */
     public void send(String method, Object... args) {
-        if (hubConnectionState != HubConnectionState.CONNECTED) {
-            throw new RuntimeException("The 'send' method cannot be called if the connection is not active.");
+        hubConnectionStateLock.lock();
```

Comment:
I don't see how acquiring this lock covers all cases. What after releasing the lock, the hubConnectionState changes from Connected to Disconnected or visa virsa?

Reply:
I was thinking about this last night actually. We'd have to lock over the entire method body pretty much to make any guarantees.